### ISSUE TITLE
Outgoing middleware design broken

### ISSUE THREAD

**Issue by rogeralsing** [AsynkronIT/protoactor-dotnet#101]
Currently the outgoing middleware has all the parts it needs internally.
_but_, the `Process.SendUserMessage` has the signature: `(PID pid, object message, PID sender)`
which does not allow us for sending outgoing headers.
And if we try to send a MessageEnvelope, things get double wrapped if there is a sender as the LocalProcess checks for sender and wraps the current message in another MessageEnvelope.

TLDR;

We should change the signature of `SendUserMessage` to just `(PID pid, object message)` and do the envelope wrapping already in `Request`

**Comment by cpx86** [comment]
Do we have the same issue in Go? I.e. is this a porting issue or was the solution broken from the start?

**Comment by rogeralsing** [comment]
Yes, same issue there

**Comment by cpx86** [comment]
Yeah maybe we should have a new "protoactor" repo for cross-cutting concerns. That could also be used for core documentation that's not specific to any language.

**Comment by cpx86** [comment]
Or we can just repurpose the "gam-web" repo - rename it to "protoactor" and use it for both the website and for general/meta issue tracking.

**Comment by rogeralsing** [comment]
relates to #127

**Comment by rogeralsing** [comment]
This has been fixed
